### PR TITLE
README Improvements

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Motivation
- Improve the README in order to make it more clear and with an easier understanding.
- Solve the issue: https://github.com/integr8ly/installation/issues/171

## What
- Add tables for the envs
- Add versions for min requirements
- Add README index
- Add the step `3. Check the connection with the OpenShift cluster` before starting the installation. 
- Add Troubleshooting in order to cover issues as for example https://github.com/integr8ly/installation/issues/171
- Add missing steps regards the OAuth GitHub Setup
- Add how to create env with RHPDS valid just for partners and redhatters
## Why
- Better understanding
- Avoid issues and mistakes
- Users are able to install it following the README 

## How
By editions in the README.md file. 

## Checklist:

- [X] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [X] Finished task
- [ ] TODO

## Additional Notes
The final result can be checked [here](https://github.com/integr8ly/installation/blob/c2ca9f6e6b92ccb6f49051bd829c2cfe772638a4/README.md).

### CODE REVIEW

**Comment by camilamacedo86** [comment]
@pb82, hi could you check it. It is a simple one.

**Comment by pb82** [comment]
👀

File: README.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ These products include:
 
 ## Prerequisites
 
-* Ansible v2.6
-* Openshift Container Platform v3.10
+* Ansible >= v2.6
+* Openshift Container Platform >= v3.10 < 3.11
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply README.md]: 👍 Change done. I thought that it was not working with the 3.11

File: README.md [hunk 2]
```diff
@@ -55,6 +55,8 @@ ansible_user=evals
 master.evals.example.com
 ```
 
+**NOTE:** It is possible add the variable `ansible_ssh_private_key_file` for the master host when the ssh connection requires a public key.(E.g`ansible_ssh_private_key_file=~/.ssh/ocp-workshop.pem`)
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply README.md]: 👍 done

**Comment by camilamacedo86** [comment]
@pb82 changes done. Please, feel free to merge it. I am still without permission so far.

File: README.md [hunk 3]
```diff
@@ -14,10 +47,12 @@ These products include:
...
+# Installing Integreatly
+
 ## Prerequisites
 
-* Ansible v2.6
-* Openshift Container Platform v3.10
+* Ansible >= v2.6
+* Openshift Container Platform >= v3.10
 * Openshift CLI (OC) v3.10
 * SSH Access to Openshift master(s)
```
Comment (odra) [inline README.md]: I think it's worth mentioning that the master ssh user needs to be able to run `sudo` commands

**Comment by odra** [comment]
This looks very good, just left a comment.

File: README.md (hunk 3 above)
Comment (camilamacedo86) [reply README.md]: Really tks for the suggestion. will add this before doing a merge. Please, could you approve it?

**Pull request merged**